test/Parser.js

```diff
@@ -132,6 +132,36 @@ describe('Parser', () => {
     ).to.equal(`.user [href="https://css-tricks.com?a=1&b=2"]{color:red;}`)
   })
 
+  test('escaped chars in selector identifiers', () => {
+    expect(
+      stylis(`
+        &.B\\&W{color:red;}
+        &.\\@example\\.com{color:blue;}
+        &.owner\\/founder{color:green;}
+     `)
+    ).to.equal(`.user.B\\&W{color:red;}.user.\\@example\\.com{color:blue;}.user.owner\\/founder{color:green;}`)
+  });
+
+  test('escaped hex digits in selector identifiers', () => {
+    expect(
+      stylis(`
+        &.B\\26W{color:red;}
+        &.B\\000026W{color:green;}
+        &.B\\26 W{color:blue;}
+        &.endsWith\\0000A9  a.childNode{color:green;}
+        &.endsWith\\AE  a.childNode{color:yellow;}
+     `)
+    ).to.equal([
+      '.user.B\\26W{color:red;}',
+      '.user.B\\000026W{color:green;}',
+      '.user.B\\26 W{color:blue;}',
+      // next rules are important because the hex digits terminating space
+      // is not the same as a combinator space
```

Comment:
Demonstration that the double-space is important: https://jsfiddle.net/6gsr7qv4/

Reply:
Is `\000026A `valid unicode wouldn't it be `\26A `?

Reply:
It's valid. From the recommendation I linked to in the PR description, https://www.w3.org/TR/css-syntax-3/#escaping

> EXAMPLE 3
> An identifier with the value "&B" could be written as `\26 B` or `\000026B`.

Reply:
... although the trailing `A` isn't part of the hex code, since the hex code is "one to six hex digits". In both what commented and what's in the recommendation's EXAMPLE 3, the hex code is `\26` (or `\000026`) and the `A` or `B` which follows is _not_ part of the hex code.

Reply:
... as a matter of fact, because `A` is a hex digit, `\000026A` and `\26A` are different - one is `&A` and the other is `ɪ`, because with neither leading `0`s nor a whitespace after the hex code, the `A` is treated as part of the hex code!

So:

* `Q\000026A` -> `Q&A`
* `Q\000026 A` -> `Q&A`
* `Q\26 A` -> `Q&A`
* `Q\26A` -> `Qɪ`

Reply:
So all unicode sequences need two spaces to separate them i.e `Q \000026 A` would need to be `Q \000026  A` with two spaces to yield `Q & A`?

Reply:
I think github collapses spaces for inline code snippets, and I can't follow if you're trying to use a className of `Q & A` (which wouldn' t work because classNames are delimited by spaces in the HTML) or trying to use space as a descendant combinator.

If you had something like `id="Q & A"` you'd need to escape the spaces just the same as the ampersand, and since the escaping slash is not a hex digit you wouldn't need a double space.

If you wanted to have a space which is a combinator after the hex digits, yes you'd need a double space.

Examples: https://jsfiddle.net/sz67trap/

Reply:
No meant selecting `\000026 A` where a is another selector i.e `\000026 header` wouldn't that now treat it as `unicode` + `header` without a space between them?

Reply:
Yes, this rule:

```
.Q\000026 header {
    color: red;
}
```

Targets this element:

```
<div class="Q&header">
```

To target this:

```
<div class="Q&">
    <header>
```

You would need a double space in the selector rule - one to terminate the hex code and one to use as a combinator:

```
.Q\000026  header {
    color: blue;
}
```

See also https://jsfiddle.net/4heovdau/

------

This is accounted for in the tests:
* a single space after 6-0r-fewer hex digits needs to be preserved: https://github.com/thysultan/stylis.js/pull/260/files#diff-4cafb040dcdb12664448ab294d801e96bb1378853e3a2a4337d280b1d3de7f89R158
* a double space after 6-or-fewer hex digits needs to be preserved: https://github.com/thysultan/stylis.js/pull/260/files#diff-4cafb040dcdb12664448ab294d801e96bb1378853e3a2a4337d280b1d3de7f89R170-R171
* a double space after a non-hex escape sequence is just a combinator and can be collapsed into a single whitespace char: https://github.com/thysultan/stylis.js/pull/260/files#diff-4cafb040dcdb12664448ab294d801e96bb1378853e3a2a4337d280b1d3de7f89R141
* a double space after the 7th hex digit is no longer a hex code terminating space but just a run of the mill combinator space which can be collapsed: https://github.com/thysultan/stylis.js/pull/260/files#diff-4cafb040dcdb12664448ab294d801e96bb1378853e3a2a4337d280b1d3de7f89R172